### DISCUSSION TITLE
Showing Trial Number in Experiment

### DISCUSSION THREAD

**Discussion by mshastry2022** [jspsych/jsPsych#874] [Q&A] [answered]
Hello, 

I am trying to find a way to display the trial number for participants when they take part in an experiment. I have 10 trials, where each trial consists of a fixation cross, a survey-text where a participant enters in a number, and a response depending on their number. In the survey-text portion, how can I say "Trial #_" on each of the trials, and how can I update it every trial?

I tried using on_start and on_finish, and I tried to use a for loop (var i = 1; i <= 10; i++). But I keep getting stuck in my code. 

I have some portions of my code. Any help will be appreciated!!

Thank you in advance!!
Madhulika

```js
var survey = {
    type: 'survey-text',
    questions: [
      {prompt: 'Enter a number:', columns: 3, name: 'Number', required: true},
   ],
    preamble: "<img src='numberline.png'></img>",
    data: {task: 'guess-number'}
}

var practice_procedure = {
    timeline: [fixation, survey, practice_feedback],
    repetitions: 10,
}
        
var display_practice = {
    type: "html-keyboard-response",
    stimulus: function() {
    return practice_earn.toString() + " dollars is what you have gained.</p><p>Press any key to continue to the next page.</p>";
    }
}
timeline.push(practice_procedure, display_practice);
```

**Comment by jodeleeuw** [comment, marked as answer]
One way to do this is to count how many trials in the data match `{task: 'guess-number'}` to figure out which trial you are currently on. You could do this in the preamble parameter:

```js
var survey = {
    type: 'survey-text',
    questions: [
      {prompt: 'Enter a number:', columns: 3, name: 'Number', required: true},
   ],
    preamble: function(){
	var trial_number = jsPsych.data.get().filter({task: 'guess-number'}).count() + 1;
	return "<p>Trial "+trial_number+" of 10</p>"+
		"<img src='numberline.png'></img>"
    },
    data: {task: 'guess-number'}
}
```
